application/F3DCLIOptionsTools.h

```diff
@@ -0,0 +1,241 @@
+/**
+ * @class   F3DCLIOptionsTools
+ * @brief   A namespace to recover CLI Options
+ *
+ */
+
+#ifndef F3DCLIOptionsTools_h
+#define F3DCLIOptionsTools_h
+
+#include "F3DConfig.h"
+
+#include <map>
+#include <string>
+#include <variant>
+#include <vector>
+
+namespace F3DCLIOptionsTools
+{
+struct CLIOption
+{
+  std::string LongName;
+  std::string ShortName;
+  std::string HelpText;
+  std::string ValueHelper;
+  std::string ImplicitValue;
+};
+
+// TODO not working ?
+static inline const std::vector<std::string> OrderedCLIGroups = { "Applicative", "General",
+  "Material", "Window", "Scientific visualization", "Camera", "Raytracing", "PostFX (OpenGL)",
+  "Testing" };
+
+// clang-format off
+static inline const std::multimap<std::string, CLIOption> CLIOptions = {
```

Comment:
Instead of having a `std::vector<std::string>` for the ordered groups, and a `std::multimap<std::string, CLIOption>` for the options, what about creating a structure like:
```cpp
struct CLIGroup
{
  std::string GroupName;
  std::vector<CLIOption>;
};
```
And have a `std::vector<CLIGroup> OrderedCLIGroups`?

Reply:
thatd work, however cxxopts does not respect my vector order (for now), ill take a look

Reply:
indeed better